src/test/java/apoc/algo/WeaklyConnectedComponentsTest.java

```diff
@@ -0,0 +1,126 @@
+package apoc.algo;
+
+import apoc.util.TestUtil;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Map;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.test.TestGraphDatabaseFactory;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class WeaklyConnectedComponentsTest {
+	
+	private static GraphDatabaseService db;
+	// TODO refactor in multiple execute statements dirty workaround
+	public static final String CC_GRAPH =
+            "CREATE (:Node:Record { name : 'A' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'B' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'C' }) "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'D' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'E' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'F' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'G' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'H' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'I' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'J' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'K' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'L' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'M' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'N' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'Z' })  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node),(c:Node)  "+
+            "WHERE a.name = 'C' AND b.name = 'D' AND c.name = 'E'  "+
+            "CREATE (a)-[:LINK]->(b)-[:LINK]->(c)-[:LINK]->(a)  "+   
```

Comment:
You can just spell your relationship-creates out
CREATE (c:Node {name:'C'})
CREATE (c)-[:LINK]->(d:Node {name:'D'})-[:LINK]->(e:Node {name:'E'})-[:LINK]->(c)
CREATE (f:Node {name:'F'})
CREATE (f)-[:LINK]->(g:Node {name:'G')-[:LINK]->(f)
... etc